--- a/src/mimonscreenplugins.cpp
+++ b/src/mimonscreenplugins.cpp
@@ -19,9 +19,9 @@
 #include <QDebug>
 
 #include <algorithm>
-#include <tr1/functional>
+#include <functional>
 
-using namespace std::tr1::placeholders;
+using namespace std::placeholders;
 
 namespace
 {
@@ -116,10 +116,10 @@ bool MImOnScreenPlugins::isEnabled(const QString &plugin) const
 
     std::remove_copy_if(mEnabledSubViews.begin(), mEnabledSubViews.end(),
                         std::back_inserter(mEnabledAndAvailableSubViews),
-                        std::tr1::bind(&MImOnScreenPlugins::isSubViewUnavailable, this, _1));
+                        std::bind(&MImOnScreenPlugins::isSubViewUnavailable, this, _1));
 
     return std::find_if(mEnabledAndAvailableSubViews.begin(), mEnabledAndAvailableSubViews.end(),
-                        std::tr1::bind(equalPlugin, _1, plugin)) != mEnabledAndAvailableSubViews.end();
+                        std::bind(equalPlugin, _1, plugin)) != mEnabledAndAvailableSubViews.end();
 }
 
 bool MImOnScreenPlugins::isSubViewEnabled(const SubView &subView) const
@@ -136,7 +136,7 @@ QList<MImOnScreenPlugins::SubView> MImOnScreenPlugins::enabledSubViews(const QSt
 {
     QList<MImOnScreenPlugins::SubView> result;
     std::remove_copy_if(mEnabledSubViews.begin(), mEnabledSubViews.end(),
-                        std::back_inserter(result), std::tr1::bind(notEqualPlugin, _1, plugin));
+                        std::back_inserter(result), std::bind(notEqualPlugin, _1, plugin));
     return result;
 }
 
--- a/tests/ut_mimpluginmanagerconfig/ut_mimpluginmanagerconfig.cpp
+++ b/tests/ut_mimpluginmanagerconfig/ut_mimpluginmanagerconfig.cpp
@@ -32,7 +32,7 @@
 #include "mattributeextensionmanager.h"
 #include "msharedattributeextensionmanager.h"
 
-using namespace std::tr1;
+//using namespace std;
 
 typedef QSet<Maliit::HandlerState> HandlerStates;
 Q_DECLARE_METATYPE(HandlerStates);
--- a/tests/ut_mimpluginmanagerconfig/ut_mimpluginmanagerconfig.h
+++ b/tests/ut_mimpluginmanagerconfig/ut_mimpluginmanagerconfig.h
@@ -15,7 +15,7 @@
 
 #include "mimserveroptions.h"
 #include "mimsettingsqsettings.h"
-#include <tr1/memory>
+#include <memory>
 
 #include <QtTest/QtTest>
 #include <QObject>
