diff --git a/configs/appcfg/profile_template b/configs/appcfg/profile_template
index 851bd39..322b1a5 100755
--- a/configs/appcfg/profile_template
+++ b/configs/appcfg/profile_template
@@ -25,7 +25,7 @@
 command -v firefox >/dev/null && export BROWSER=firefox
 
 # Prepopulate Subreddits menu with custom subreddits
-#export SXMO_SUBREDDITS="postmarketos pinephoneOfficial pinephone unixporn"
+#export SXMO_SUBREDDITS="chimeralinux pinephoneOfficial pinephone unixporn"
 
 # Change the default terminal command
 export TERMCMD="sxmo_terminal.sh"
diff --git a/configs/default_hooks/sxmo_hook_block_suspend.sh b/configs/default_hooks/sxmo_hook_block_suspend.sh
index a5a7f82..b7aab1c 100755
--- a/configs/default_hooks/sxmo_hook_block_suspend.sh
+++ b/configs/default_hooks/sxmo_hook_block_suspend.sh
@@ -56,6 +56,10 @@ hotspot_active() {
 	nmcli -t c show --active | grep -q ^Hotspot
 }
 
+# TODO: expand to allow blocking on other tcp port activity as well
+# e.g. ss -Htn '( src :22 or src :25565 )'
+# or possibly even check any actively used port at all: ss -Htun
+# (also what about UDP ports?)
 ssh_connected() {
 	[ -n "$(ss -Htn src :22)" ]
 }
diff --git a/configs/default_hooks/sxmo_hook_logout.sh b/configs/default_hooks/sxmo_hook_logout.sh
index 15247e1..61df517 100755
--- a/configs/default_hooks/sxmo_hook_logout.sh
+++ b/configs/default_hooks/sxmo_hook_logout.sh
@@ -5,9 +5,9 @@
 # "cool off" before we restart wm, but otherwise everything *should* die with the
 # wm
 superctl stop sxmo_soundmonitor
-superctl stop wireplumber
-superctl stop pipewire-pulse
-superctl stop pipewire
+#superctl stop wireplumber
+#superctl stop pipewire-pulse
+#superctl stop pipewire
 
 # give pipewire time
 sleep 5s
diff --git a/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh b/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh
index f48fcef..70aa07c 100755
--- a/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh
+++ b/configs/default_hooks/sxmo_hook_restart_modem_daemons.sh
@@ -26,6 +26,9 @@ daemon_start() {
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			doas systemctl start "$1"
 			;;
+		chimera)
+			doas dinitctl start "$1"
+			;;
 	esac
 }
 
@@ -46,6 +49,9 @@ daemon_stop() {
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			doas systemctl stop "$1"
 			;;
+		chimera)
+			doas dinitctl stop "$1"
+			;;
 	esac
 }
 
@@ -62,6 +68,9 @@ daemon_isrunning() {
 			[ "$1" = "modemmanager" ] && set -- ModemManager
 			systemctl is-active --quiet "$1"
 			;;
+		chimera)
+			dinitctl --system --quiet is-started "$1"
+			;;
 	esac
 }
 
@@ -75,6 +84,10 @@ daemon_exists() {
 			systemctl status "$1" > /dev/null 2>&1
 			[ $? -ne 4 ]
 			;;
+		chimera)
+			# TODO: should user services be accounted for here? -> " || [ -f /etc/dinit.d/user/"$1" ]"
+			[ -f /etc/dinit.d/"$1" ]
+			;;
 	esac
 }
 
diff --git a/configs/default_hooks/sxmo_hook_start.sh b/configs/default_hooks/sxmo_hook_start.sh
index 50fa86e..1e220be 100755
--- a/configs/default_hooks/sxmo_hook_start.sh
+++ b/configs/default_hooks/sxmo_hook_start.sh
@@ -30,8 +30,9 @@ if [ -z "$SXMO_NO_AUDIO" ]; then
 		superctl start pulseaudio
 	elif [ "$(command -v pipewire)" ]; then
 		# pipewire-pulse will start pipewire
-		superctl start pipewire-pulse
-		superctl start wireplumber
+		#superctl start pipewire-pulse
+		#superctl start wireplumber
+		:
 	fi
 
 	# monitor for headphone for statusbar
diff --git a/configs/default_hooks/sxmo_hook_wallpaper.sh b/configs/default_hooks/sxmo_hook_wallpaper.sh
index 82a5274..5e4df2a 100755
--- a/configs/default_hooks/sxmo_hook_wallpaper.sh
+++ b/configs/default_hooks/sxmo_hook_wallpaper.sh
@@ -8,11 +8,12 @@
 # This script is executed once, and must output the wallpaper to display
 
 installed_wallpapers() {
+	# TODO: do something for chimera here? it has default wallpapers
 	xdg_data_path wallpapers 0 "\0" | xargs -r0I{} find "{}" -name "$SXMO_OS.*"
 }
 
 sxmo_wallpaper() {
-	xdg_data_path sxmo/background.jpg
+	xdg_data_path backgrounds/chimera/bg-l.svg
 }
 
 all_wallpapers() {
diff --git a/configs/profile.d/sxmo_init.sh b/configs/profile.d/sxmo_init.sh
index fbe5ce5..a5d354d 100644
--- a/configs/profile.d/sxmo_init.sh
+++ b/configs/profile.d/sxmo_init.sh
@@ -94,6 +94,7 @@ _sxmo_load_environments() {
 
 	# The user can already force a $SXMO_DEVICE_NAME value in ~/.profile
 	if [ -z "$SXMO_DEVICE_NAME" ]; then
+		# TODO: /sys/devices/virtual/dmi/id/product_name ("V3", "MS-7C95")
 		if [ -e /proc/device-tree/compatible ]; then
 			SXMO_DEVICE_NAME="$(tr -c '\0[:alnum:].,-' '_' < /proc/device-tree/compatible |
 				tr '\0' '\n' | head -n1)"
diff --git a/scripts/appscripts/sxmo_reddit.sh b/scripts/appscripts/sxmo_reddit.sh
index f908729..2c6b566 100755
--- a/scripts/appscripts/sxmo_reddit.sh
+++ b/scripts/appscripts/sxmo_reddit.sh
@@ -6,7 +6,7 @@
 # shellcheck source=scripts/core/sxmo_common.sh
 . sxmo_common.sh
 
-[ -z "$SXMO_SUBREDDITS" ] && SXMO_SUBREDDITS="pine64official pinephoneofficial unixporn postmarketos linux"
+[ -z "$SXMO_SUBREDDITS" ] && SXMO_SUBREDDITS="pine64official pinephoneofficial unixporn chimeralinux linux"
 
 menu() {
 	SUBREDDIT="$(
diff --git a/scripts/core/sxmo_bluetoothmenu.sh b/scripts/core/sxmo_bluetoothmenu.sh
index 8aaadc5..cf535b8 100755
--- a/scripts/core/sxmo_bluetoothmenu.sh
+++ b/scripts/core/sxmo_bluetoothmenu.sh
@@ -59,6 +59,9 @@ _restart_bluetooth() {
 		arch|archarm|nixos)
 			doas systemctl restart bluetooth
 			;;
+		chimera)
+			doas dinitctl restart bluetoothd
+			;;
 	esac
 }
 
diff --git a/scripts/core/sxmo_bluetoothtoggle.sh b/scripts/core/sxmo_bluetoothtoggle.sh
index f670f61..1c4b7f5 100755
--- a/scripts/core/sxmo_bluetoothtoggle.sh
+++ b/scripts/core/sxmo_bluetoothtoggle.sh
@@ -18,6 +18,9 @@ on() {
 			systemctl start bluetooth
 			systemctl enable bluetooth
 			;;
+		chimera)
+			dinitctl enable bluetoothd
+			;;
 	esac
 }
 
@@ -31,6 +34,9 @@ off() {
 			systemctl stop bluetooth
 			systemctl disable bluetooth
 			;;
+		chimera)
+			dinitctl disable bluetoothd
+			;;
 	esac
 	rfkill block bluetooth
 }
diff --git a/scripts/core/sxmo_contactmenu.sh b/scripts/core/sxmo_contactmenu.sh
index 5c6ba8a..75579b4 100755
--- a/scripts/core/sxmo_contactmenu.sh
+++ b/scripts/core/sxmo_contactmenu.sh
@@ -40,7 +40,7 @@ editcontactname() {
 
 	if ! printf %s "$PICKED" | grep -q "^Old name: "; then
 		newcontact="$oldnumber	$PICKED"
-		sed -i "s/^$oldnumber	$oldname$/$newcontact/" "$SXMO_CONTACTFILE"
+		sed -i '' "s/^$oldnumber	$oldname$/$newcontact/" "$SXMO_CONTACTFILE"
 		set -- "$newcontact"
 	fi
 
@@ -68,7 +68,7 @@ editcontactnumber() {
 	newcontact="$PICKED	$oldname"
 
 	# reverse them
-	sed -i "s/^$number	$name$/$newcontact/" "$SXMO_CONTACTFILE"
+	sed -i '' "s/^$number	$name$/$newcontact/" "$SXMO_CONTACTFILE"
 	editcontact "$oldname	$PICKED"
 }
 
@@ -84,7 +84,7 @@ deletecontact() {
 	)"
 
 	# reverse them
-	printf %s "$PICKED" | grep -q "Yes" && sed -i "/^$number	$name$/d" "$SXMO_CONTACTFILE"
+	printf %s "$PICKED" | grep -q "Yes" && sed -i '' "/^$number	$name$/d" "$SXMO_CONTACTFILE"
 }
 
 editcontact() {
diff --git a/scripts/core/sxmo_mmsdconfig.sh b/scripts/core/sxmo_mmsdconfig.sh
index b923c78..772bee9 100755
--- a/scripts/core/sxmo_mmsdconfig.sh
+++ b/scripts/core/sxmo_mmsdconfig.sh
@@ -56,7 +56,7 @@ editfile() {
 		VALUE="$(printf %s "$CHOICE" | cut -d= -f2-)"
 		NEWVALUE="$(valuemenu "$KEY" "$VALUE")"
 
-		sed -i "$FILE" -e "s|^$CHOICE$|$KEY=$NEWVALUE|"
+		sed -i '' -e "s|^$CHOICE$|$KEY=$NEWVALUE|" "$FILE"
 	done
 }
 
diff --git a/scripts/core/sxmo_networks.sh b/scripts/core/sxmo_networks.sh
index d0327bf..5c10406 100755
--- a/scripts/core/sxmo_networks.sh
+++ b/scripts/core/sxmo_networks.sh
@@ -10,7 +10,7 @@
 
 set -e
 
-VPNDEVICE="$(nmcli con show --active | grep -E 'wireguard|vpn' | awk '{ print $1 }' | paste -s -d '|')"
+VPNDEVICE="$(nmcli con show --active | grep -E 'wireguard|vpn' | awk '{ print $1 }' | paste -s -d '|' -)"
 
 nofail() {
 	"$@" || return 0
diff --git a/scripts/core/sxmo_status.sh b/scripts/core/sxmo_status.sh
index 2e50b28..f0ac0cc 100755
--- a/scripts/core/sxmo_status.sh
+++ b/scripts/core/sxmo_status.sh
@@ -102,7 +102,8 @@ show() {
 	_sorted_components_name | while read -r id; do
 		tr '\n' ' ' < "$ROOT/$id"
 		printf " "
-	done | head -c -1 | tr '\n' '\0' | xargs -0 -n1 printf "%s\n"
+	done | sed 's/.$//' | tr '\n' '\0' | xargs -0 -n1 printf "%s\n"
+	# ^ FIXED head: illegal byte count -- -1
 	free
 }
 
diff --git a/scripts/core/sxmo_timezonechange.sh b/scripts/core/sxmo_timezonechange.sh
index f680a96..b590a30 100755
--- a/scripts/core/sxmo_timezonechange.sh
+++ b/scripts/core/sxmo_timezonechange.sh
@@ -24,6 +24,15 @@ change_systemd() {
 	echo "Timezone changed ok"
 }
 
+change_fallback() {
+	echo "Changing timezone to $1"
+
+	doas sxmo_set_timezone.sh "$1"
+	sxmo_hook_statusbar.sh time
+
+	echo "Timezone changed ok"
+}
+
 menu() {
 	zoneinfo_dir=$(xdg_data_path zoneinfo)
 	T="$(
@@ -39,9 +48,11 @@ if [ $# -gt 0 ]; then
 	trap "read -r" EXIT
 	set -e
 
+	# FIXME: call change_fallback once confirming $@ is e.g. Europe/Helsinki
 	case "$SXMO_OS" in
 		alpine|postmarketos) change_alpine "$@";;
 		arch|archarm|debian) change_systemd "$@";;
+		chimera) echo "Change the timezone via ln -sf ../usr/share/zoneinfo/$@ /etc/localtime";;
 		nixos) echo "Change the timezone in configuration.nix with time.timeZone = \"[timezone]\"";;
 		*) echo "Changing the timezone isn't implemented on your distro yet";;
 	esac
diff --git a/scripts/core/sxmo_upgrade.sh b/scripts/core/sxmo_upgrade.sh
index 9607b1a..f69d720 100755
--- a/scripts/core/sxmo_upgrade.sh
+++ b/scripts/core/sxmo_upgrade.sh
@@ -37,7 +37,7 @@ update_nixos() {
 sxmo_wakelock.sh lock sxmo_upgrading infinite
 
 case "$SXMO_OS" in
-	alpine|postmarketos) update_apk;;
+	alpine|postmarketos|chimera) update_apk;;
 	arch|archarm) update_pacman;;
 	nixos) update_nixos;;
 esac
diff --git a/scripts/core/sxmo_vvmdconfig.sh b/scripts/core/sxmo_vvmdconfig.sh
index 6cfab48..412e0d7 100755
--- a/scripts/core/sxmo_vvmdconfig.sh
+++ b/scripts/core/sxmo_vvmdconfig.sh
@@ -52,7 +52,7 @@ editfile() {
 		VALUE="$(printf %s "$CHOICE" | cut -d= -f2-)"
 		NEWVALUE="$(valuemenu "$KEY" "$VALUE")"
 
-		sed -i "$FILE" -e "s|^$CHOICE$|$KEY=$NEWVALUE|"
+		sed -i '' -e "s|^$CHOICE$|$KEY=$NEWVALUE|" "$FILE"
 	done
 }
 
diff --git a/scripts/modem/sxmo_modemaudio.sh b/scripts/modem/sxmo_modemaudio.sh
index 41acf03..f435396 100755
--- a/scripts/modem/sxmo_modemaudio.sh
+++ b/scripts/modem/sxmo_modemaudio.sh
@@ -3,6 +3,7 @@
 # shellcheck source=scripts/core/sxmo_common.sh
 . sxmo_common.sh
 
+# TODO: silence if callaudio{cli,d} missing?
 ca_dbus_get_prop() {
 	dbus-send --session --print-reply --dest=org.mobian_project.CallAudio \
 		--reply-timeout=2500 /org/mobian_project/CallAudio org.freedesktop.DBus.Properties.Get \
diff --git a/scripts/modem/sxmo_modemtext.sh b/scripts/modem/sxmo_modemtext.sh
index cc9487a..a69576f 100755
--- a/scripts/modem/sxmo_modemtext.sh
+++ b/scripts/modem/sxmo_modemtext.sh
@@ -105,7 +105,7 @@ sendtextmenu() {
 			# Remove Attachment
 			"$icon_cls $icon_att"*)
 				FILE="$(printf %s "$CONFIRM" | awk -F' :: ' '{print $2}')"
-				sed -i "\|$FILE|d" "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt"
+				sed -i '' "\|$FILE|d" "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt"
 				if [ ! -s "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt" ] ; then
 					rm "$SXMO_LOGDIR/$NUMBER/draft.attachments.txt"
 				fi
