--- a/cmake/modules/Findwlroots.cmake
+++ b/cmake/modules/Findwlroots.cmake
@@ -23,7 +23,7 @@
 #==============================================================================
 
 find_package(PkgConfig)
-pkg_check_modules(PKG_wlroots QUIET wlroots)
+pkg_check_modules(PKG_wlroots QUIET wlroots-0.18)
 
 set(wlroots_VERSION ${PKG_wlroots_VERSION})
 set(wlroots_INCLUDE_DIRS ${PKG_wlroots_INCLUDE_DIRS})
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -219,6 +219,7 @@ if (${wlroots_VERSION} VERSION_GREATER_EQUAL 0.18)
   set(WLR_HAVE_NEW_PIXEL_COPY_API 1)
   set(WLR_HAVE_BACKEND_CREATE_WITH_LOOP 1)
   set(WLR_HAVE_WL_POINTER_ENUMS 1)
+  set(WLR_HAVE_UNIFIED_TEXTURE_FORMATS_GETTER 1)
 endif()
 
 find_package(X11)
--- a/como/base/config-como.h.cmake
+++ b/como/base/config-como.h.cmake
@@ -21,6 +21,7 @@
 #cmakedefine01 WLR_HAVE_NEW_PIXEL_COPY_API
 #cmakedefine01 WLR_HAVE_BACKEND_CREATE_WITH_LOOP
 #cmakedefine01 WLR_HAVE_WL_POINTER_ENUMS
+#cmakedefine01 WLR_HAVE_UNIFIED_TEXTURE_FORMATS_GETTER
 
 #if HAVE_BREEZE_DECO
 #define BREEZE_KDECORATION_PLUGIN_ID "${BREEZE_KDECORATION_PLUGIN_ID}"
--- a/como/render/backend/wlroots/egl_backend.h
+++ b/como/render/backend/wlroots/egl_backend.h
@@ -71,7 +71,11 @@ public:
         wayland::init_egl(*this, data);
 
         if (this->hasExtension(QByteArrayLiteral("EGL_EXT_image_dma_buf_import"))) {
+#if WLR_HAVE_UNIFIED_TEXTURE_FORMATS_GETTER
+            auto const formats_set = wlr_renderer_get_texture_formats(backend.renderer, WLR_BUFFER_CAP_DMABUF);
+#else
             auto const formats_set = wlr_renderer_get_dmabuf_texture_formats(backend.renderer);
+#endif
             auto const formats_map = get_drm_formats<Wrapland::Server::drm_format>(formats_set);
 
             dmabuf = std::make_unique<Wrapland::Server::linux_dmabuf_v1>(
