From d1308320aa13a808aca288784364405acbf1a282 Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Sun, 22 Sep 2024 11:41:10 +0300
Subject: [PATCH 08/14] halium-lxc-exec.sh: use lxc-attach from env

---
 src/halium-lxc-exec.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/halium-lxc-exec.sh b/src/halium-lxc-exec.sh
index 8600575..83eb66d 100755
--- a/src/halium-lxc-exec.sh
+++ b/src/halium-lxc-exec.sh
@@ -24,4 +24,4 @@ found_path=$(whereis -b -B ${ANDROID_SEARCH_PATH} -f ${TARGET_BINARY} | awk '{ p
 unset LD_PRELOAD
 
 # Finally execute
-exec /usr/bin/lxc-attach -n ${LXC_CONTAINER_NAME} -- ${found_path#${LXC_CONTAINER_PATH}} "${@}" \;
+exec lxc-attach -n ${LXC_CONTAINER_NAME} -- ${found_path#${LXC_CONTAINER_PATH}} "${@}" \;
-- 
2.46.0

