From b7a7e9b45d83c8ce0abf68b4f75f1d5fe830866a Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Fri, 20 Sep 2024 23:27:06 +0300
Subject: [PATCH 13/14] [CHIMERA] dinit readiness

---
 src/android-service.sh | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/android-service.sh b/src/android-service.sh
index 13048d9..8bd2e79 100644
--- a/src/android-service.sh
+++ b/src/android-service.sh
@@ -23,8 +23,21 @@ current_status() {
 	getprop init.svc.${service_service}
 }
 
+dinit_ready() {
+	[ "${DINIT_SERVICE}" ] || return
+
+	# Notify dinit we're done (service started)
+	echo > /dev/fd/${ANDROID_SERVICE_READY_FD}
+
+	# Then simply block until the service is no longer running
+	WAITFORSERVICE_VALUE_NOT=running waitforservice init.svc.${service_service}
+}
+
 start() {
-	[ "$(current_status)" = "running" ] && return 0
+	if [ "$(current_status)" = "running" ]; then
+		dinit_ready
+		return 0
+	fi
 
 	# Start operation is weird since it's kickstarted by Android's
 	# init - thus we assume that if ${ANDROID_SERVICE_STAMP} doesn't
@@ -41,6 +54,7 @@ start() {
 
 	# Once we return, create the stamp file
 	touch ${ANDROID_SERVICE_STAMP}
+	dinit_ready
 }
 
 stop() {
-- 
2.46.0

