From 500784922a190fc9d6049e510454c06f50ee883a Mon Sep 17 00:00:00 2001
From: Jami Kettunen <jami.kettunen@protonmail.com>
Date: Sat, 28 Sep 2024 20:45:24 +0300
Subject: [PATCH 14/14] Makefile: Install android_ldd wrapper

---
 src/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Makefile b/src/Makefile
index 2db78cb..7417f6e 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -20,6 +20,7 @@ SYMLINKS = \
 	android_stop \
 	android_start \
 	android_kill \
+	android_ldd \
 	logcat \
 	lshal
 
-- 
2.46.0

