--- a/serenity/AK/Assertions.cpp
+++ b/serenity/AK/Assertions.cpp
@@ -10,8 +10,10 @@
 #include <AK/StringBuilder.h>
 #include <AK/StringView.h>
 
-#if (defined(AK_OS_LINUX) && !defined(AK_OS_ANDROID)) || defined(AK_LIBC_GLIBC) || defined(AK_OS_BSD_GENERIC) || defined(AK_OS_SOLARIS) || defined(AK_OS_HAIKU)
+#if 0
 #    define EXECINFO_BACKTRACE
+#else
+#    define PRINT_ERROR(s) (void)fputs((s), stderr)
 #endif
 
 #if defined(AK_OS_ANDROID) && (__ANDROID_API__ >= 33)
